lib/Extractor/Candidates.cpp

```diff
@@ -215,6 +216,31 @@ Inst *ExprBuilder::buildGEP(Inst *Ptr, gep_type_iterator begin,
   return Ptr;
 }
 
+void ExprBuilder::markExternalUses (Inst *I) {
+  std::map<Inst *, unsigned> UsesCount;
+  std::vector<Inst *> Stack;
+  Stack.push_back(I);
+  while(!Stack.empty()) {
+    Inst* T = Stack.back();
+    Stack.pop_back();
+    if (I != T) {
+      if (UsesCount.find(T) == UsesCount.end())
+        UsesCount[T] = 1;
+      else
+        UsesCount[T]++;
+    }
+    for (auto i: T->Ops) {
```

Comment:
I think it's best to avoid using both "i" and "I" in the same function, and also please use initial capital letters for local vars, and also we usually put spaces on both sides of the ":" in this context